cache-filter/src/filter/http.rs

```diff
@@ -87,13 +87,24 @@ pub struct CacheFilter {
 }
 
 impl HttpContext for CacheFilter {
-    fn on_http_request_headers(&mut self, context_id: usize) -> Action {
+    fn on_http_request_headers(&mut self, _: usize) -> Action {
         let mut request_data = match self.get_request_data() {
             Ok(data) => data,
             Err(e) => {
-                debug!("ctxt {}: fetching request data failed: {}", e, context_id);
+                debug!(self.context_id, "fetching request data failed: {}", e);
+
                 // Send back local response for not providing relevant request data
-                self.send_http_response(401, vec![], None);
+                if cfg!(feature = "visible_logs") {
+                    #[cfg(feature = "visible_logs")]
+                    {
+                        let (key, val) =
+                            crate::log::visible_logs::get_logs_header_pair(self.context_id);
+                        self.send_http_response(401, vec![(key.as_ref(), val.as_ref())], None);
+                    }
+                } else {
+                    self.send_http_response(401, vec![], None);
+                }
```

Comment:
If you use `cfg!` then you should not use `#[cfg]` right below, as that is implied by `cfg!`. This macro is just an easier way to deal with more than two configurations when the code excludes branches depending on which configurations are enabled. In this particular case, I'd just change this to be:
```suggestion
                let body = if cfg!(feature = "visible_logs") {
                    let (key, val) =
                        crate::log::visible_logs::get_logs_header_pair(self.context_id);
                    vec![(key.as_ref(), val.as_ref())]
                    self.send_http_response(401, vec![(key.as_ref(), val.as_ref())], None);
                } else {
                    vec![]
                };
                self.send_http_response(401, body, None);
```

Reply:
I was trying something similar but got errors saying "visible_logs crate not found". This is due the fact that when `visible_logs` flag is off, that crate is not available but line 
```Rust
crate::log::visible_logs::get_logs_header_pair(self.context_id);
``` 
is still present in the source code for compilation.

Reply:
Builds fine but again clippy thinks it's alive.

Reply:
The `does not remove any code` is a sentence for you to be aware that the other branches must still be correct code, rather than potentially uncompilable stuff - the behaviour is otherwise identical, since the compiler will optimize away `if true`/`if false`, which is what these end up compiling into.